### PR TITLE
PullRequest for Issue128: Make the pv configuration file configurable for AMDSPV

### PR DESCRIPTION
**Pull request by XuetaoLiu-S**
https://github.com/acquaman/AcquamanDataServer/issues/128

Make the configuration file configuable

### CODE REVIEW

**Comment by XuetaoLiu-S** [comment]
oops... not reviewed yet :bomb: :boom: . @iainworkman is working on PR now :(

File: source/AMDSMainCommon.h [hunk 1]
```diff
@@ -14,10 +14,14 @@ namespace AMDSMain_Common {
...
-	void printUsage(const QString &appName)
+	void printUsage(const QString &appName, const QString &errMsg="")
 	{
+		if (errMsg != "") {
+			qDebug() << "\nERROR: " << errMsg << "\n";
+		}
+
 		qDebug() << "\nUsages:\n"
-				 << QString("	%1 --intf=[interface name] [--port=[portnumber]]\n").arg(appName)
+				 << QString("	%1 --intf=[interface name] [--port=[portnumber]] [--cfg=[cfgFileName]]\n").arg(appName)
```
Comment (iainworkman) [inline source/AMDSMainCommon.h:24]: The bracketing here seems to suggest that the --cfg is optional, but it isn't it doesn't seem to be in usage (line 39). Is this correct?

File: source/application/SGM/AMDSCentralServerSGMPV.cpp [hunk 2]
```diff
@@ -49,14 +49,14 @@ void AMDSCentralServerSGMPV::onNewPVDataReceived(const QString &bufferName, AMDS
 	}
 }
 
-void AMDSCentralServerSGMPV::initializeConfiguration()
+void AMDSCentralServerSGMPV::initializeConfiguration(const QString &configurationFileName)
 {
 	// initialize the configurations of the PVs
 	QRegExp rx("(\\,|\\t)"); //RegEx for ',' or '\t'
 	int totalParameters = 6;
 //	int dataTypeParamIndex = totalParameters - 1;
```
Comment (iainworkman) [inline source/application/SGM/AMDSCentralServerSGMPV.cpp:57]: Is this commented code still required?

**Comment by iainworkman** [comment]
Couple of minor issues, neither of which affect the functionality. 
Time Estimate: 5 minutes.

File: source/application/SGM/AMDSCentralServerSGMPV.cpp (hunk 2 above)
Comment (XuetaoLiu-S) [inline source/application/SGM/AMDSCentralServerSGMPV.cpp:57]: Yes. it is still required by far

File: source/AMDSMainCommon.h (hunk 1 above)
Comment (XuetaoLiu-S) [inline source/AMDSMainCommon.h:24]: yes, --cfg is optional. Ln39 gives the cfgFileName a default value, which doesn't hurt anything. But in ln58, we had a general check for all the required parameters.

But for AMDSPV, this parameter is required. I had a specific check in https://github.com/acquaman/AcquamanDataServer/blob/3a12ce60ac5da83f6baf8be774790a8931146cfe/source/AMDSMainSGMPV.cpp#L20.

**Pull request merged**